src/main/java/ch/geowerkstatt/ilivalidator/extensions/functions/PolylinesOverlapIoxPlugin.java

```diff
@@ -0,0 +1,145 @@
+package ch.geowerkstatt.ilivalidator.extensions.functions;
+
+import ch.interlis.ili2c.metamodel.PathEl;
+import ch.interlis.ili2c.metamodel.Viewable;
+import ch.interlis.iom.IomObject;
+import ch.interlis.iom_j.itf.impl.jtsext.geom.CompoundCurve;
+import ch.interlis.iox.IoxException;
+import ch.interlis.iox_j.jts.Iox2jtsext;
+import ch.interlis.iox_j.validator.Value;
+import com.vividsolutions.jts.geom.IntersectionMatrix;
+import com.vividsolutions.jts.index.strtree.STRtree;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public final class PolylinesOverlapIoxPlugin extends BaseInterlisFunction {
+    private static final Map<HasEqualLinePartKey, Boolean> HAS_EQUAL_LINE_PART_CACHE = new HashMap<>();
+
+    @Override
+    public String getQualifiedIliName() {
+        return "GeoW_FunctionsExt.PolylinesOverlap";
+    }
+
+    @Override
+    protected Value evaluateInternal(String validationKind, String usageScope, IomObject contextObject, Value[] arguments) {
+        Value argObjects = arguments[0];
+        Value argPath = arguments[1];
+
+        if (argObjects.isUndefined()) {
+            return Value.createSkipEvaluation();
+        }
+
+        if (argObjects.getComplexObjects() == null) {
+            return Value.createUndefined();
+        }
+
+        Collection<IomObject> polylineObjects;
+
+        if (argPath.isUndefined()) {
+            polylineObjects = argObjects.getComplexObjects();
+        } else {
+            Viewable contextClass = EvaluationHelper.getContextClass(td, contextObject, argObjects);
+            if (contextClass == null) {
+                throw new IllegalStateException("unknown class in " + usageScope);
+            }
+
+            PathEl[] attributePath = EvaluationHelper.getAttributePathEl(validator, contextClass, argPath);
+            polylineObjects = EvaluationHelper.evaluateAttributes(validator, argObjects, attributePath);
+        }
+
+        Collection<IomObject> inputObjects = argObjects.getComplexObjects();
+        boolean hasObjectIds = inputObjects.stream().anyMatch(o -> o.getobjectoid() != null);
+        if (!hasObjectIds) {
+            List<CompoundCurve> lines = convertToJTSLines(polylineObjects);
+            return new Value(hasEqualLinePart(lines));
+        }
+
+        List<String> objectIds = inputObjects.stream().map(IomObject::getobjectoid).collect(Collectors.toList());
+        HasEqualLinePartKey key = new HasEqualLinePartKey(objectIds, argPath.isUndefined() ? null : argPath.getValue());
+
+        boolean hasOverlap = HAS_EQUAL_LINE_PART_CACHE.computeIfAbsent(key, k -> {
```

Comment:
Ich hatte noch vergessen, die Resultate zu cachen.
Macht es Sinn, dass für den Cache die Objekt-Ids und der Attributpfad verwendet werden?
Die übliche Verwendung ist vermutlich in einem Set-Constraint mit `GeoW_FunctionsExt.PolylinesOverlap(ALL, 'Geometrie');`, da sollten alle Objekte eine Id haben.